rust-toolchain.toml

```diff
@@ -0,0 +1,2 @@
+[toolchain]
+channel = "nightly-2024-11-01"
```

Comment:
Can you elaborate on why nightly is needed? If we need to rely on nightly-only features, my understanding is that would force consumers of this crate to use nightly too. But it is definitely possible that I am overthinking the impact downstream, so curious to hear your thoughts!

Reply:
Good question, I think most people will use nightly because of : https://book.leptos.dev/deployment/binary_size.html so they can rebuild the standard library with optimisation, also I did that because the cargo project itself use nightly for its CI, but maybe nightly is not fundamentally needed and once we stabilise the crate we can think about using stable rust

Reply:
Fair , there are many optimizations in nightly -- maybe I am overblowing the impact if this crate doesn't use nightly code logic features. Thanks for the explanation

Reply:
Oh no I think your concern is valid!
Actually, you should even open an issue so we look back at it once we start stabilizing the crate.